### ISSUE TITLE
Dorado confuses different methylation types with modified base calling models

### ISSUE THREAD

**Issue by alexcritschristoph** [nanoporetech/dorado#322]
I'm not sure if this is a dorado issue or a remora issue, but I've been running Dorado with the latest modified base models like so:

```
dorado basecaller dna_r10.4.1_e8.2_400bps_sup@v4.2.0 $library_name --emit-moves --modified-bases-models dna_r10.4.1_e8.2_400bps_sup@v4.2.0_5mC@v2,dna_r10.4.1_e8.2_400bps_sup@v4.2.0_6mA@v2 > $library_name.d3.bam
```

I've been running this on data from E. coli which is dcm+ (CCWGG (5mC) and dam+ GATC (6mA). What I see in this data is that the "A" position in the CC[A|T]GG motif will also be called as methylated with fairly high confidence by dorado. My understanding is this is biologically inaccurate. Note that the model will correctly call the C in CCWGG is methylated as well.

My thinking is this is happening is because the nearby methylated C creates an abnormal signal that causes the adjacent A to look modified. You could imagine if a classifier is trained to distinguish 6mA methylated data from unmethylated data, without ever having 5mC data, this confusion could arise. However, this is suboptimal behavior, because essentially the base caller is calling a base as modified that is not. 

In general the modified base models are very helpful, but just flagging this potential behavior as it could result in confusing results for some users. I am also not sure what the extent of the issue is beyond my particular dataset.

**Comment by marcus1487** [comment]
We are aware of this issue in the all-context modified base models and are working hard to address this in the model training data. The next release of the all-context models should see a marked improvement in this behavior. Thank you for flagging this up. I will post here when these new models are released.

**Comment by mattloose** [comment]
This is quite a significant issue - are you saying that some of the models are known to be problemattic? It would be good to be clear on which models are currently stable?

**Comment by marcus1487** [comment]
We have continued to develop and test the all-context models upon release and have noticed that other modified bases nearby can introduce false positive calls as noted by @alexcritschristoph. This does not happen at all contexts or modified bases, but does occur at some non-trivial rate. We will continue to improve these models after this first release and the next release will contain specific efforts to address nearby 5mC and 6mA bases. Issues will still likely remain for other modified bases not presented at training time. We will continue to develop these models and address issues.

These model will be stable in the sense that we will continue to support all-context 5mC and 6mA model releases with future basecaller releases, but the characteristics of one modified base in the context of another modified base may change in future iterations of these models. We will also aim to increase overall accuracy of all modified base models.

**Comment by mattloose** [comment]
Thanks Marcus - so the 5mC model alone (for example) is not affected and can be considered  robust (within the parameters that work can be ongoing and improving)?

Matt

**Comment by marcus1487** [comment]
The released 5mC all-context model was not trained with canonical cytosines neighboring a 6mA base, yet we have not observed higher false positives for 5mC near 6mA bases in limited test sets. We cannot guarantee that other contexts outside our available samples will not produce false positive 5mC calls. Similarly we cannot guarantee that new modified bases will not shift the signal in a way that the model will interpret as a false positive/negative 5mC. As stated, we will continue to attempt to improve the models and include more robust training data, but we will never be able to train and test all modified bases in the context of all other modified bases.

**Comment by ywang285** [comment]
Hi Alex and Marcus, thank you for the useful discussion. I also saw some high 5mCG regions have significantly high 5hmCG and 6mA calling, when I was using dna_r10.4.1_e8.2_400bps_sup@v4.2.0_5mCG_5hmCG@v2 (5 kHz) to call 5mCG and 5hmCG, as well as using dna_r10.4.1_e8.2_400bps_sup@v4.2.0_6mA@v2 (5 kHz) to call 6mA modifications on DNA. My understanding is that both 5hmCG and 6mA are likely false positive at these regions but 5mCG is most likely true positive? May I ask the expected time for the new model to release? Thanks!

**Comment by marcus1487** [comment]
There are two separate issues here. For the 6mA calls near native human CG sites (presumably methylated), this is part of the issue discussed in this thread that we are attempting to address in the next model release. We don't have a concrete timeline for this model release, but this is one of the top priorities for the modified base team internally so this is a short term target (currently developing major Remora 3.0 and associated Betta release to facilitate these updates for release as well).

For the higher 5hmC rates in CG regions this is explicitly modeled by the `dna_r10.4.1_e8.2_400bps_sup@v4.2.0_5mCG_5hmCG@v2`, so for these sites we would expect the error rates to be as presented in the London Calling Tech update. The expected error rate for the 5mC+5hmC model on synthetically printed ground truth strands are found in the plot below. Note that these quoted error rate are using the 10% filtering which is the default in our recommended `modkit pileup` [tool for downstream processing](https://github.com/nanoporetech/modkit). If you are seeing levels of 5hmC significantly higher than these error rates then it is possible that this is real 5hmC. There are obviously many other possibilities of unknown modified bases which would effect these calls as well, which is why we only quote accuracy numbers against synthetically printed ground truth strands.

![image](https://github.com/nanoporetech/dorado/assets/13755444/3bd070cd-9f3e-4f43-b516-c16531b3d65b)

**Comment by ywang285** [comment]
Thank you so much for the clarification!
